json/src/pointer.rs

```diff
@@ -0,0 +1,141 @@
+//! JSON Pointer
+//!
+//! This module presents JSON Pointers through the `Pointer` trait and implements it for the `Value` type.
+//!
+//! For more information read [RFC6901](https://tools.ietf.org/html/rfc6901).
+
+use value::Value;
+
+/// Provides the `pointer` method for locating values within an object using a string path.
+pub trait Pointer {
```

Comment:
Yes, please move these methods directly onto the type. If we ever get a more concrete use case for splitting them off, we'll do that then